### PR TITLE
[MPS] Fix `.item()` for multi-dim scalar

### LINKED ISSUE #107867: tensor.item() fails on mps device
### 🐛 Describe the bug

`.item()` fails with an error when evaluated on a tensor of shape `[1, 1]` on an mps device:

```
>>> torch.tensor([[1]]).to("mps").item()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: Expected dst.dim() >= src.dim() to be true, but got false.  (Could this error message be improved?  If so, please report an enhancement request to PyTorch.)
```

There's no problem when the tensor has shape `[1]`:
```
>>> torch.tensor([1]).to("mps").item()
1
```

Also no problem when not using mps:
```
>>> torch.tensor([[1]]).item()
1
```

### Versions

```
Collecting environment information...
PyTorch version: 2.1.0.dev20230823
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 13.5 (arm64)
GCC version: Could not collect
Clang version: 13.1.6 (clang-1316.0.21.2.3)
CMake version: version 3.25.2
Libc version: N/A

Python version: 3.11.2 (main, Jul 14 2023, 21:44:29) [Clang 13.1.6 (clang-1316.0.21.2.3)] (64-bit runtime)
Python platform: macOS-13.5-arm64-arm-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Apple M1 Max

Versions of relevant libraries:
[pip3] numpy==1.25.1
[pip3] torch==2.1.0.dev20230823
[pip3] torchaudio==2.1.0.dev20230823
[pip3] torchvision==0.16.0.dev20230823
[conda] Could not collect
```

cc @kulinseth @albanD @malfet @DenisVieriu97 @razarmehr @abhudev

### PR DESCRIPTION
**Pull request by malfet**
By refactoring `_local_scalar_dense_mps` to use `_empty_like` to allocate CPU tensor.
Also, print a more reasonable error message when dst dim is less than src in mps_copy_

This fixes regression introduced by https://github.com/pytorch/pytorch/pull/105617 and adds regression test.

### <samp>🤖 Generated by Copilot at abd06e6</samp>

> _Sing, O Muse, of the valiant deeds of the PyTorch developers_
> _Who strive to improve the performance and usability of tensors_
> _And who, with skill and wisdom, fixed a bug in the MPS backend_
> _That caused confusion and dismay to many a user of `item()`_

Fixes https://github.com/pytorch/pytorch/issues/107867

### CODE REVIEW

File: aten/src/ATen/native/mps/operations/Copy.mm [hunk 1]
```diff
@@ -305,6 +304,9 @@ void copy_blit_mps(void* dst, const void* src, size_t size) {
         break;
       }
     }
+  } else {
+    // dst.dim() can be smaller than src.dim() when scalars are copied
+    TORCH_CHECK(dst.numel() == 1, "Destination has ", dst.dim(), " dims, which is fewer than src ", src.dim());
```
Comment (albanD) [inline aten/src/ATen/native/mps/operations/Copy.mm:309]: Shouldn't this check that src.numel() == 1 as well?

Comment (malfet) [reply aten/src/ATen/native/mps/operations/Copy.mm:309]: How about adding `TORCH_CHECK_INTERNAL(src.numel() == dst.numel())` at the start of the function? (but after resize) (On the other hand, check that number of elements for src and dst is the same should already be handled in `Copy.cpp`, shouldn't it?)

Comment (albanD) [reply aten/src/ATen/native/mps/operations/Copy.mm:309]: Did broadcasting already happen by then?

Comment (albanD) [reply aten/src/ATen/native/mps/operations/Copy.mm:309]: Actually is that true?
```
import torch

a = torch.rand(1)
b = torch.rand(1, 1, 1)

a.copy_(b)

```
This fails for me?
Is our cuda copy not consistent with our cpu copy?

Comment (malfet) [reply aten/src/ATen/native/mps/operations/Copy.mm]: @albanD well, `output with shape [1] doesn't match the broadcast shape [1, 1, 1]` exception is triggered for `tensor.copy_` method invocation after my change as well, which I guess part of the meta operator call, which item somehow bypasses..

**Review by albanD — APPROVED** [review]
SGTM!

**Pull request closed without merging**